lib/spack/spack/test/conftest.py

```diff
@@ -670,6 +672,12 @@ def writer_key_function():
         )
     return _impl
 
+
+@pytest.fixture()
+def mock_gnupghome(tmpdir, monkeypatch):
+    monkeypatch.setattr(spack.util.gpg, 'GNUPGHOME', str(tmpdir.join('gpg')))
```

Comment:
@tgamblin I think if we merge this refactor and #14594 is rebased on top, you just need to add:
```python
monkeypatch.setattr(spack.util.gpg.Gpg, '_gpg', None)
```
to make the PR work. Didn't try it out though.

Reply:
@alalazo: this looks great -- thanks for the cleanup.